internal/controllers/objectsets/objectsetphases_reconciler.go

```diff
@@ -76,12 +76,25 @@ type phaseReconciler interface {
 func (r *objectSetPhasesReconciler) Reconcile(
 	ctx context.Context, objectSet genericObjectSet,
 ) (res ctrl.Result, err error) {
-	activeObjects, probingResult, err := r.reconcile(ctx, objectSet)
+	controllerOf, probingResult, err := r.reconcile(ctx, objectSet)
 	if err != nil {
 		return res, err
 	}
+	objectSet.SetStatusControllerOf(controllerOf)
+
+	inTransition := r.isObjectSetInTransition(objectSet, controllerOf)
+	if inTransition {
```

Comment:
Dont we also want to make sure that if an objectset is in transition we dont mark it as available?